### PR TITLE
feat(redux-module-mercury): move mercury status out of spark module

### PR DESCRIPTION
**Pull request by adamweeks**
This makes our redux-spark module depend on the mercury module, but it sets things up so I can do the reconnect logic.

### CODE REVIEW

File: packages/node_modules/@ciscospark/redux-module-mercury/src/index.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import {Map} from 'immutable';
+
+export const SET_CONNECTING = `mercury/SET_CONNECTING`;
+export const SET_DISCONNECTING = `mercury/SET_DISCONNECTING`;
+export const SET_CONNECTED = `mercury/SET_CONNECTED`;
+export const SET_DISCONNECTED = `mercury/SET_DISCONNECTED`;
```
Comment (bzang) [inline packages/node_modules/@ciscospark/redux-module-mercury/src/index.js]: Doesn't look like these disconnect action names aren't being used anywhere? Are these gonna be used when everything gets migrated over?

File: packages/node_modules/@ciscospark/redux-module-mercury/src/index.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+export const initialState = new Map({
+  connected: false,
+  connecting: false
+});
```
Comment (bzang) [inline packages/node_modules/@ciscospark/redux-module-mercury/src/index.js]: Can we set these into a status object to be consistent with the spark module?

File: packages/node_modules/@ciscospark/redux-module-mercury/src/index.js (hunk 1 above)
Comment (adamweeks) [inline packages/node_modules/@ciscospark/redux-module-mercury/src/index.js]: Oops, leftovers.

File: packages/node_modules/@ciscospark/redux-module-mercury/src/index.js (hunk 2 above)
Comment (adamweeks) [inline packages/node_modules/@ciscospark/redux-module-mercury/src/index.js]: Sure.

**Comment by adamweeks** [comment]
@bzang changes requested have been added

**Pull request merged**